elasticsearch_dsl/search_base.py

```diff
@@ -352,20 +352,34 @@ def __getitem__(self, n):
             # If negative slicing, abort.
             if n.start and n.start < 0 or n.stop and n.stop < 0:
                 raise ValueError("Search does not support negative slicing.")
-            # Elasticsearch won't get all results so we default to size: 10 if
-            # stop not given.
-            s._extra["from"] = n.start or 0
-            s._extra["size"] = max(
-                0, n.stop - (n.start or 0) if n.stop is not None else 10
-            )
-            return s
+            slice_start = n.start
+            slice_stop = n.stop
         else:  # This is an index lookup, equivalent to slicing by [n:n+1].
             # If negative index, abort.
             if n < 0:
                 raise ValueError("Search does not support negative indexing.")
-            s._extra["from"] = n
-            s._extra["size"] = 1
-            return s
+            slice_start = n
+            slice_stop = n + 1
+
+        old_from = s._extra.get("from")
+        old_to = (old_from or 0) + s._extra["size"] if "size" in s._extra else None
```

Comment:
I initially read this as `old_to = (old_from or 0) + (s._extra["size"] if "size" in s._extra else None)`. We could use parents to make this more obvious, but arguable the expression is complex enough to *not* use a ternary expressions?

```suggestion
        old_to = None
        if "size" in s._extra:
            old_to = (old_from or 0) + s._extra["size"]
```

You can even use the walrus operator if you feel fancy:

```suggestion
        old_to = None
        if old_size := s._extra.get(size):
            old_to = (old_from or 0) + old_size
```